core/services/ocr2/plugins/generic/oraclefactorydb.go

```diff
@@ -0,0 +1,135 @@
+package generic
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/pkg/errors"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+type oracleFactoryDb struct {
+	// The ID is used for logging and error messages
+	// A single standard capabilities spec can instantiate multiple oracles
+	// TODO: NewOracle should take a unique identifier for the oracle
+	specID               int32
```

Comment:
I'm not sure specID is correct here: maybe rename to instance ID + accept an ID in the oracle args?